src/Microsoft.AspNet.Mvc.Core/ActionDescriptorProviderContext.cs

```diff
@@ -3,15 +3,10 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.AspNet.Mvc
+namespace Microsoft.AspNet.Mvc.Core
```

Comment:
> I used .Core for things in the root folder of MVC that shouldn't be user visible by default. (Like I mentioned above more are needed though)

I followed this change, we should think about doing something better, and organizing these classes further. I like the idea of cleaning it up, I just don't like `.Core`. IMO it would be better to separate out namespace cleanup from this change. That usually will come with a lot of churn and moving files around the tree as well.